### PR TITLE
Message Forwarding

### PR DESCRIPTION
**Pull request by t3chguy**
for #2597 and #3592

soft-depends on vector-im/riot-web#3688

### CODE REVIEW

**Comment by ara4n** [comment]
woo, this looks very cool. will play with it a bit to see how it feels.

**Comment by turt2live** [comment]
Forwarding to an encrypted room with new devices has an interesting side effect:
image

The first 'neat' was before devices were introduced. The second was after. There was no devices prompt for the second message, and the status bar didn't pop up until navigating away and back to the room.

**Comment by t3chguy** [comment]
thanks @turt2live for the feedback
I'm not sure what the best way to get the UDE to catch this is, whether wrapping the sending of the message in a q() so that it happens on next tick

**Comment by t3chguy** [comment]
also noticed another bug with the rhs

**Comment by ara4n** [comment]
lgtm modulo feedback in https://github.com/vector-im/riot-web/pull/3688

File: src/components/views/rooms/ForwardMessage.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            case KeyCode.ESCAPE:
+                this.props.onCancelClick();
+                break;
+        }
+    },
+
+    render: function() {
+        return (
+            <div className="mx_ForwardMessage">
+                <h1>Please select the destination room for this message</h1>
```
Comment (t3chguy) [inline src/components/views/rooms/ForwardMessage.js]: needs i18n

Comment (t3chguy) [inline src/components/views/rooms/ForwardMessage.js]: matrix-org/matrix-react-sdk/pull/951

**Pull request merged**